routing/simple.html

```diff
@@ -0,0 +1,240 @@
+              }
+            };
+
+            fc.features.push(origin_feature);
+            fc.features.push(destination_feature);
+            fc.features.push(route_feature);
+          }); // legs
+
+          // alternates are lighter
+          color *= Math.floor(color_scale);
+          color_scale = 1;
+
+        }); // routes
+      }); // responses
+
+      //give back the feature collection
+      return fc;
+    }
+
+    function render(req_rep) {
+      const {requests, responses} = req_rep;
+      previous_geojson = geojson;
+      //turn the result into geojson
+      fc = build_geojson(responses)
+      //make a leaflet geojson object
+      geojson = L.geoJson(fc, {
+        style: (feature) => { return feature.properties; },
+        pointToLayer: (feature, ll) => { return new L.CircleMarker(ll, feature.properties); }
+      });
+      //clear the previous
+      if(previous_geojson != null) {
+        previous_geojson.removeFrom(map);
+      }
+      //disappear the input
+      inputs.forEach( (input) => {
+        input.removeFrom(map);
+      });
+      //render the geojson
+      var added = geojson.addTo(map);
+      //fit it in view
+      map.fitBounds(added.getBounds(), { maxZoom: 19 });
+      //update text box and anchor
+      document.getElementById('request_body').value = JSON.stringify(requests);
+      window.history.pushState(null, null, window.location.pathname + '#' + JSON.stringify(requests));
+    }
+
+
+    ///////////////////// EVENTS ////////////////////////
+
+    //build a location object for the request
+    map.on('click', (event) => {
+      //clear this if its not null
+      if(geojson != null) {
+        inputs = [];
+        geojson.removeFrom(map);
+        geojson = null;
+        document.getElementById('request_body').value = '{"costing":"auto"}';
+      }
+
+      //make the location
+      let body = document.getElementById('request_body').value;
+      try{
+        body = JSON.parse(body);
+      }
+      catch(e){
+        alert('Invalid Json Request');
+      }
+
+      if (!('locations'  in body)) {
+        body['locations'] = [];
+      }
+      body['locations'].push({ lat: event.latlng.lat, lon: event.latlng.lng });
+      document.getElementById('request_body').value = JSON.stringify(body);
+      window.history.pushState(null, null, window.location.pathname + '#' + JSON.stringify(body));
+
+      //mark where they clicked
+      input = new L.CircleMarker(event.latlng, { color: '#0000bb', radius: 5, fillOpacity: 0.75 });
+      input.addTo(map);
+      inputs.push(input);
+    });
+
+    //button press callback to fire off the request to the server and render the result
+    document.getElementById('route').onclick = async (event) => {
+      const results = await fetch(document.getElementById('request_body').value);
+      render(results);
+    };
+
+    //Check if we should initialize from anchor
+    $(document).ready(async () => {
```

Comment:
when the page loads, if there was an anchor we yank it out here and fetch the requests it contained